File: lib/features/savings_goals/domain/models/savings_goal.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:savvy/core/constants/financial_enums.dart';

part 'savings_goal.freezed.dart';
part 'savings_goal.g.dart';

@freezed
abstract class SavingsGoal with _$SavingsGoal {
  const factory SavingsGoal({
    required String id,
    required String title,
    required double targetAmount,
    @Default(0.0) double currentAmount,
    DateTime? targetDate,
    required SavingsCategory category,
    @Default('#D97706') String colorHex,
    @Default('target') String iconName,
    @Default(GoalStatus.active) GoalStatus status,
    required DateTime createdAt,
  }) = _SavingsGoal;

  factory SavingsGoal.fromJson(Map<String, dynamic> json) =>
      _$SavingsGoalFromJson(json);
}

File: SavvyApp/Packages/Core/SavvyFoundation/Sources/SavvyFoundation/Models/SavingsGoal.swift
import Foundation

public struct SavingsGoal: FirestoreEntity {
    public let id: String
    public var title: String
    public var targetAmount: Decimal
    public var currentAmount: Decimal
    public var targetDate: Date?
    public var category: SavingsCategory
    public var colorHex: String
    public var iconName: String
    public var status: GoalStatus
    public var isDeleted: Bool
    public let createdAt: Date

    public init(
        id: String = UUID().uuidString,
        title: String,
        targetAmount: Decimal,
        currentAmount: Decimal = 0,
        targetDate: Date? = nil,
        category: SavingsCategory = .goal,
        colorHex: String = "#D97706",
        iconName: String = "target",
        status: GoalStatus = .active,
        isDeleted: Bool = false,
        createdAt: Date = Date()
    ) {
        self.id = id
        self.title = title
        self.targetAmount = targetAmount
        self.currentAmount = currentAmount
        self.targetDate = targetDate
        self.category = category
        self.colorHex = colorHex
        self.iconName = iconName
        self.status = status
        self.isDeleted = isDeleted
        self.createdAt = createdAt
    }
}
